Refactored SoftwareVersionUIViewSet to use UI Components

sirisha-bode opened the pull request:
<!--
    Thank you for your interest in contributing to Nautobot! Please note
    that our contribution policy recommends that a feature request or bug
    report be opened for approval prior to filing a pull request. This
    helps avoid wasting time and effort on something that we might not
    be able to accept.

    Please indicate the relevant feature request or bug report below.
-->
# Closes #<ISSUE NUMBER GOES HERE>
# What's Changed

Overview of Accommodations Made:

* Converted the basic ObjectFieldsPanel to display all model fields in the left half of the detail view.
* Added a custom StatsPanel on the right half to visualize statistics related to software_version, aggregating data from SoftwareImageFile, Device, InventoryItem, and VirtualMachine.

# Screenshots
| View            | Before                                                                                     | After                                                                                       |
|-----------------|-------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------|
| List View       |![image](https://github.com/user-attachments/assets/edb324a6-b143-44c4-9f7b-40aee8c1422c)|![image](https://github.com/user-attachments/assets/fd7ac4f2-171d-42fc-9d7e-ed66a5488f41)|
| Detail View     |![image](https://github.com/user-attachments/assets/6464d307-162a-4e0a-aab4-5696e5097760)|![image](https://github.com/user-attachments/assets/00c634a8-1d57-41bc-b911-6771f79c9ae6)|
| Create View     |![image](https://github.com/user-attachments/assets/720c9e16-284a-4681-8fb7-e92fc654e72b)![image](https://github.com/user-attachments/assets/4783fe6e-3e7d-4913-ba16-2abba7cf79dc)|![image](https://github.com/user-attachments/assets/6552ae40-306a-4c5c-a5a5-2fb530e237fa)![image](https://github.com/user-attachments/assets/e4549154-1786-4bf5-955a-a1bbcd2a7c54)|
| Update View     |![image](https://github.com/user-attachments/assets/90bf1aee-e998-49bb-998a-d6fcab3b888d)![image](https://github.com/user-attachments/assets/aad60fca-44a0-44e2-b41b-9dbb51e4f9ef)|![image](https://github.com/user-attachments/assets/bfae24b8-e611-4125-82bd-072191d3dc6e)![image](https://github.com/user-attachments/assets/6f846b21-99b6-492c-ad55-3322063795f5)|
| Delete View     |![image](https://github.com/user-attachments/assets/04663ae0-5997-43ee-bcd8-bbdd935525e4)![image](https://github.com/user-attachments/assets/516abcfe-4b91-44ed-9fc0-2327d67e2d2d)|![image](https://github.com/user-attachments/assets/9e56c0cb-7ffe-45d2-bac5-87b6104d424d)![image](https://github.com/user-attachments/assets/b57b1d61-6ae7-431e-a93c-ab92810955f4)|
| Bulk Edit View  |![image](https://github.com/user-attachments/assets/26ace42b-a738-4364-a393-1049a584a9ed)![image](https://github.com/user-attachments/assets/78602471-aed3-4fe7-8a3a-f49cd260027a)|![image](https://github.com/user-attachments/assets/17bb82d3-2605-48cf-85f0-db53f848925f)![image](https://github.com/user-attachments/assets/923994eb-55a3-4d3f-9198-3931aef9d051)|
| Bulk Delete View |![image](https://github.com/user-attachments/assets/25cc9113-01ea-452d-885f-baf44f4ed0fb)![image](https://github.com/user-attachments/assets/745c93e7-168d-45fe-8dab-037528e13b91)|![image](https://github.com/user-attachments/assets/0fd36ade-71aa-4e41-9941-87626eee1faa)![image](https://github.com/user-attachments/assets/a0009cca-64ba-4c92-bd01-d78ecb85fcae)|

# TODO
<!--
    Please feel free to update todos to keep track of your own notes for WIP PRs.
-->
- [ ] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/core/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

Sridevi-NTC commented on nautobot/dcim/templates/dcim/softwareversion_retrieve.html:
```diff
@@ -1,92 +1,2 @@
-                            <td>Documentation URL</td>
-                            <td>
-                                {% if object.documentation_url %}
-                                <a href="{{ object.documentation_url }}" target="_blank">{{ object.documentation_url }}</a>
```
I believe `hyperlinked_field ` doesn't support `target="_blank"`, and we can't use value transforms either ([Reference](https://github.com/nautobot/nautobot/blob/develop/nautobot/core/ui/object_detail.py#L1171)). Let's reach out to the core team to decide if the Documentation URL should open in a new tab. Based on their feedback, you can implement custom logic.

@itdependsnetworks, @glennmatthews, @HanlinMiao  your input is appreciated

itdependsnetworks replied:
My vote would be to not worry about it, all things considering, I would rather be consistent, and I know for myself, I just command click to open in new tab.

Sridevi-NTC commented on nautobot/dcim/views.py:
```diff
@@ -4368,6 +4369,31 @@ class SoftwareVersionUIViewSet(NautobotUIViewSet):
+                section=SectionChoices.LEFT_HALF,
+                fields="__all__",
+                value_transforms={
+                    "status": [helpers.hyperlinked_object_with_color],
+                    "alias": [helpers.placeholder],
+                    "platform": [helpers.hyperlinked_object],
+                    "release_date": [helpers.placeholder],
+                    "end_of_support_date": [helpers.placeholder],
+                    "long_term_support": [helpers.render_boolean],
+                    "pre_release": [helpers.render_boolean],
```
They are already handled in framework, helpers are not required

sirisha-bode replied:
Updated as requested, should be resolve, let me know if I missed anything

sirisha-bode commented:
Overview of Accommodations Made:
1. Converted the basic ObjectFieldsPanel to display all model fields in the left half of the detail view.
2. Added a custom StatsPanel on the right half to visualize statistics related to software_version, aggregating data from SoftwareImageFile, Device, InventoryItem, and VirtualMachine.

itdependsnetworks commented on nautobot/dcim/views.py:
```diff
@@ -4264,6 +4264,22 @@ class SoftwareVersionUIViewSet(NautobotUIViewSet):
+            object_detail.ObjectFieldsPanel(
+                weight=100,
+                section=SectionChoices.LEFT_HALF,
+                fields="__all__",
```
Is there a templatetag to humanize the date? 

![image](https://github.com/user-attachments/assets/458833d4-ba5f-473e-bcc4-1827563489fc)

itdependsnetworks replied:
Once this is addressed, feel free to update tag to qa-review-request

glennmatthews replied:
Make sure any approach taken here is considerate of issues like #6902 and the plan to eventually address #7018?

sirisha-bode replied:
Yes, we do have the humanize template tag, and I’ve used it before. However, I noticed that even without explicitly applying the value transformation, the UI looks the same. So, I think we might not actually need to use the humanize tag in this case. Also, I had forgotten to update the image in the detail view earlier, but that’s fixed now.

glennmatthews reviewed (commented):
Looks solid to me - what's needed still to move this out of draft status?
